naive_ralloc/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+#![feature(allocator)]
+#![feature(linkage)]
+#![no_std]
+#![allocator]
+
+#[macro_use]
+extern crate sc;
+
+use core::ptr;
+
+// fs/mmap.c
+pub unsafe fn brk(brk: usize) -> usize {
+    syscall!(BRK, brk)
+}
+
+fn allocate(size: usize) -> *mut u8 {
```

Comment:
Shouldn't this use SBRK instead of BRK?

The man page says

> brk() sets the end of the data segment to the value specified by addr, when that value is reasonable, the system has enough memory, and the process does not exceed its maximum data size (see setrlimit(2)).

> sbrk() increments the program's data space by increment bytes. Calling sbrk() with an increment of 0 can be used to find the current location of the program break. 

Which sounds like using BRK would "allocate" new memory over previously allocated memory. OTOH, SBRK would hand over disjoint memory blocks.